dfdx/tests/issue_tests.rs

```diff
@@ -0,0 +1,36 @@
+use dfdx::prelude::*;
+use std::fmt::Debug;
+
+#[test]
+fn test_issue_891() {
```

Comment:
This is a cool test 😂

Reply:
Indeed, I think it's helpful to prevent logical bugs at comp time, in case someone tries to add shapes into something that isn't one!